pkg/syncer/syncer.go

```diff
@@ -282,11 +283,28 @@ func StartSyncer(ctx context.Context, cfg *SyncerConfig, numSyncerThreads int, i
 			}
 
 			// Start and sync informer factories
+			waitForCacheSyncContext, cancelWaitForCacheSync := context.WithCancel(ctx)
+			defer cancelWaitForCacheSync()
+			var waitForCacheSyncErrorCount int
 			var cacheSyncsForAlwaysRequiredGVRs []cache.InformerSynced
+			waitForCacheSyncErrors := map[schema.GroupVersionResource]map[string]int{}
+			var errorHandlerMutex sync.Mutex
 			for _, alwaysRequiredGVR := range alwaysRequiredGVRs {
 				if informer, err := ddsifForUpstreamSyncer.ForResource(alwaysRequiredGVR); err != nil {
 					return nil, err
 				} else {
+					errors := make(map[string]int)
+					_ = informer.Informer().SetWatchErrorHandler(func(r *cache.Reflector, err error) {
+						errorHandlerMutex.Lock()
+						defer errorHandlerMutex.Unlock()
+						errStr := err.Error()
+						errors[errStr]++
+						waitForCacheSyncErrorCount++
+						if waitForCacheSyncErrorCount > 10 {
```

Comment:
why 10 and not 27 ?

Reply:
Mostly arbitrary (not too small, not too big ;-) ), but I added a comment to explain the choice.

Reply:
Fixed